subprojects/base-services/src/main/java/org/gradle/internal/hash/FileHasher.java

```diff
@@ -29,6 +31,12 @@
     HashCode hash(InputStream inputStream);
 
     /**
+     * Returns the hash of the given input stream while copying the data to the output stream.
+     * The method will not close either stream.
```

Comment:
Added. Though I think we handle closing streams in a more reliable way in general. I keep thinking about how to address this...